### PR TITLE
if update grid only contains nodata, skip the update

### PR DESCRIPTION
**Pull request by alessandrofelder**
Addresses #66 .

### CODE REVIEW

File: pycascadia/remove_restore.py [hunk 1]
```diff
@@ -65,6 +65,10 @@ def calc_diff_grid(base_grid, update_grid, diff_threshold=0.0, window_width=None
         print("Update grid is entirely outside region of interest. Skipping.")
         return None
 
+    if all_values_are_nodata(update_grid.grid):
+        print("Update grid consists entirely of no_data_values. Skipping")
```
Comment (JamieJQuinn) [inline pycascadia/remove_restore.py:69]: Just for consistency with previous messages.

```suggestion
        print("Update grid consists entirely of no_data_values. Skipping.")
```

File: pycascadia/utility.py [hunk 2]
```diff
@@ -11,6 +11,12 @@ def is_region_valid(region):
     else:
         return True
 
+def all_values_are_nodata(grid):
+    if (grid.nodatavals == grid.values).all():
+        return True
+    else:
+        return False
```
Comment (JamieJQuinn) [inline pycascadia/utility.py:18]: Seems cleaner to return the condition directly.

```suggestion
    return (grid.nodatavals == grid.values).all()
```

**Review by JamieJQuinn — APPROVED** [review]
Delicious.

**Comment by alessandrofelder** [comment]
Test failure due to newer `xarray` version documented in #68 . Merging as test failures unrelated to issue solved by this PR.

**Pull request merged**